test/general/key.js

```diff
@@ -2989,17 +2989,13 @@ module.exports = () => describe('Key', function() {
     source.subKeys = [];
     dest.subKeys = [];
     expect(dest.isPublic()).to.be.true;
-    return dest.update(source).then(() => {
-      expect(dest.isPrivate()).to.be.true;
-      return Promise.all([
-        dest.verifyPrimaryKey().then(result => {
-          expect(source.verifyPrimaryKey()).to.eventually.equal(result);
-        }),
-        dest.users[0].verify(dest.primaryKey).then(result => {
-          expect(source.users[0].verify(source.primaryKey)).to.eventually.equal(result);
-        })
-      ]);
-    });
+
+    await dest.update(source);
+    expect(dest.isPrivate()).to.be.true;
+    await expect(dest.verifyPrimaryKey()).to.be.fulfilled;
+    await expect(source.verifyPrimaryKey()).to.be.fulfilled;
+    await expect(dest.users[0].verify(dest.primaryKey)).to.be.fulfilled;
+    await expect(source.users[0].verify(source.primaryKey)).to.be.fulfilled;
```

Comment:
I'm still not really a fan of these, I would just await the functions directly, since they throw on failure it should then be clear what the test is checking. (Also elsewhere)

Reply:
My position is still that since we do not test any other result/side-effect, it's clearer to be explicit about what is being tested. I feel like leaving `await verify(...)` can look as if we just forgot about those lines.

Reply:
I think you are both right. One thing to consider is which of the two approaches produces more useful output during failures - in case there is any difference.